plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/TransformationMatrix.cpp

```diff
@@ -872,37 +896,44 @@ void TransformationMatrix::initialiseToPrim(bool readFromPrim, Transform* transf
     m_flags |= kInheritsTransform;
 
   if(matchesMayaProfile(m_xformops.begin(), m_xformops.end(), m_orderedOps.begin()))
-  {
     m_flags |= kFromMayaSchema;
-  }
-  else
-  {
-  }
+
+  // Disable push to prim if enabled, otherwise MPlug value queries and setting
+  // in the switch statement below will trigger pushing to the prim, which
+  // will create undesirable usd "over"s.
+  //
+  bool pushOriginallyEnabled = pushToPrimEnabled();
+  m_flags &= ~kPushToPrimEnabled;
 
   auto opIt = m_orderedOps.begin();
   for(std::vector<UsdGeomXformOp>::const_iterator it = m_xformops.begin(), e = m_xformops.end(); it != e; ++it, ++opIt)
   {
     const UsdGeomXformOp& op = *it;
+
+    // How many time samples does the op have?  (This will be 0 if the op only has a constant
+    // value set or if it has no value set at all.)
+    //
+    int numTimeSamples = op.GetNumTimeSamples();
+
     switch(*opIt)
     {
     case kTranslate:
       {
         m_flags |= kPrimHasTranslation;
-        if(op.GetNumTimeSamples() > 1)
-        {
+        if(numTimeSamples > 0)
           m_flags |= kAnimatedTranslation;
-        }
+
         if(readFromPrim)
         {
-          MVector tempTranslation;
-          internal_readVector(tempTranslation, op);	
+          MVector vecFromUsd;
+          internal_readVector(vecFromUsd, op);
           if(transformNode)
           {
-            MPlug(transformNode->thisMObject(), MPxTransform::translateX).setValue(tempTranslation.x);
-            MPlug(transformNode->thisMObject(), MPxTransform::translateY).setValue(tempTranslation.y);
-            MPlug(transformNode->thisMObject(), MPxTransform::translateZ).setValue(tempTranslation.z);
+            MPlug(transformNode->thisMObject(), MPxTransform::translateX).setValue(vecFromUsd.x);
+            MPlug(transformNode->thisMObject(), MPxTransform::translateY).setValue(vecFromUsd.y);
+            MPlug(transformNode->thisMObject(), MPxTransform::translateZ).setValue(vecFromUsd.z);
             m_translationTweak[0] = m_translationTweak[1] = m_translationTweak[2] = 0;
-            m_translationFromUsd = tempTranslation;
+            m_translationFromUsd = vecFromUsd;
           }
```

Comment:
I added temp variables in all the cases of the switch and did this particular rename for consistency with those.  I named them all based on the data types.  (I had originally thought about declaring all the temp variables outside the switch and reusing them to make the switch cases have a little less stuff in them, but then didn't do that to avoid having some unused variables sitting around.)

Reply:
I'm going to mark this resolved since, if we do need / want temp vars for all these branches, then standardizing the naming does make sense.  However, as I note [here](https://github.com/Autodesk/maya-usd/pull/183#discussion_r377866663), I'm still not sure that the temp vars are really necessary...